src/server/worker/spawner/map.go

```diff
@@ -1,31 +1,40 @@
 package spawner
 
-func (s *Spawner) RunMap(pachClient *client.APIClient) error {
-	logger := logs.NewMasterLogger(a.pipelineInfo)
-	// Listen for new commits, and create jobs when they arrive
-	commitIter, err := pachClient.SubscribeCommit(a.pipelineInfo.Pipeline.Name, a.pipelineInfo.OutputBranch, "", pfs.CommitState_READY)
+// forEachCommit listens for each READY output commit in the pipeline, and calls
+// the given callback once for each such commit, synchronously.
+func forEachCommit(
+	pachClient *client.APIClient,
+	pipelineInfo *pps.PipelineInfo,
+	cb func(*pfs.CommitInfo) error,
+) error {
+	commitIter, err := pachClient.SubscribeCommit(a.pipelineInfo.Pipeline.Name, pipelineInfo.OutputBranch, "", pfs.CommitState_READY)
```

Comment:
I'd recommend using `SubscribeCommitF` which is also call back based, I think it'd make this code a little bit cleaner.